src/SigasiProjectCreator/CsvParser.py

```diff
@@ -17,17 +17,23 @@ def __init__(self):
 
     def parse_file(self, csv_file, options=None):
         library_mapping = dict()
+        includes = set()
+        defines = []
         with open(csv_file, 'r') as f:
             reader = csv.reader(f, skipinitialspace=True)
             for row in reader:
                 if row:
                     library = row[0].strip()
                     path = pathlib.Path(row[1].strip()).absolute().resolve()
-                    if path in library_mapping:
+                    if library == '#include':
+                        includes.add(path)
+                    elif library == '#define':
+                        defines.append(row[1].strip())
+                    elif path in library_mapping:
                         if isinstance(library_mapping[path], list):
                             library_mapping[path].append(library)
                         else:
                             library_mapping[path] = [library_mapping[path], library]
```

Comment:
This isn't new code so I should've seen this earlier, sorry. I think it'd be better to use a "defaultdict" for the `library_mapping`. You'd be able to always have lists in there:

```python
>>> from collections import defaultdict
>>> library_mapping = defaultdict(list)
>>> path = "asdf/q"
>>> library = "work"
>>> library_mapping[path].append(library)
>>> library_mapping
defaultdict(<class 'list'>, {'asdf/q': ['work']})
>>> dict(library_mapping)
{'asdf/q': ['work']}
```

you could simplify the whole `elif path in library_mapping...`

Reply:
Implemented as suggested